tools/cross_runner/src/main.cpp

```diff
@@ -273,21 +281,25 @@ int main(int argc, const char*argv[])
 
         close_execute_reset_wait(d3d12_device.Get(), command_queue.Get(), command_allocator.Get(), command_list.Get());
         const auto descriptors_count = node->get_total_descriptor_count();
-        
-        // bind descriptor heap
-        auto descriptor_heap = create_descriptor_heap(d3d12_device.Get(), descriptors_count);
-        ID3D12DescriptorHeap* d3d12_descriptor_heaps[] = { descriptor_heap.Get() };
-        command_list->SetDescriptorHeaps(1, d3d12_descriptor_heaps);
-
-        // initalize
-        node->initialize(command_list.Get(), descriptor_heap->GetCPUDescriptorHandleForHeapStart(), descriptor_heap->GetGPUDescriptorHandleForHeapStart());
-        close_execute_reset_wait(d3d12_device.Get(), command_queue.Get(), command_allocator.Get(), command_list.Get());
 
-        // 
-        // Bind and execute the operator on the GPU.
-        // 
-        // 
-        command_list->SetDescriptorHeaps(1, d3d12_descriptor_heaps);
+        // if in staless mode, there is no need to have a descriptor heap, 
+        // otherwise we should create one to bind resources.
+        ComPtr<ID3D12DescriptorHeap> descriptor_heap;
+        if(!opts.use_stateless) 
```

Comment:
Since `use_stateless` will be part of the nodes than you can expose new virtual function in that nodes like: "is_needing_descriptor_heap()" and use it here.
That new function can have default implementation in base class which returns "true" and overwrite it in classes which have stateless support

Reply:
done. add a new virtual function, by default it will return `false`